fixed compile warning with gcc6

cromerc opened the pull request:
Now in gcc6 it gives warnings on compile when this type of code is used:
if (something == 0) 
  return 0;

The fix is simple:
if (something == 0) {
  return 0;
}

glensc commented:
include output of the actual warning

cromerc commented:
Sorry about that,
`logrotate.c: In function ‘postrotateSingleLog’:
logrotate.c:1784:5: error: this ‘if’ clause does not guard... [-Werror=misleading-indentation]
     if (!state->doRotate)
     ^~
`
This is being compiled on Arch Linux x86_64 with the latest gcc6, sysvinit, and openrc.

I am using the following configuration:
`	./configure \
		--prefix=/usr \
		--sbindir=/usr/bin \
		--mandir=/usr/share/man \
		--with-acl
`

glensc commented:
so your fix is wrong, you should fix indentation, it is seen from [github commit](https://github.com/logrotate/logrotate/pull/35/commits/b0e66573e47f44396e170f8e914592442dda61b) as well that indentation is wrong.
![image](https://cloud.githubusercontent.com/assets/199095/15427763/d6feb81c-1e9f-11e6-9fbb-40f3d4f64d35.png)

i personally never write if statement without the `{}` braces, i suggest to:
- fix indentation
- add `{}` braces to all code places having `if`-statement, not just specific place

cromerc commented:
Alright, I will fix the indentation and look for other erroneous ifs and put the {} braces around them.

glensc commented:
why did you close it. just add more commits to the same branch!

cromerc commented:
Sorry, this is my first pull request, I didn't know to leave it open. I will reopen it. I am working on the indentation now, it is a big mess.

glensc commented:
for now, just add commits and push to the same branch you added initial commit. if maintainers request to rebase or squash, that all can be still done (i can guide)

cromerc commented:
Alright, I have cleaned up the first file which is logrotate.c. I fixed all the indentations in the file, changed all the if/while/for to use curly brackets, and I took an extra step and unified all the coding styles into 1 style to make it easier to read and understand.

cromerc commented:
... the travis-ci build failed because the apt-get in the server failed. Lol.

On my machine make and make test are successful with the latest changes I pushed.

sergey-v commented on basenames.c:
```diff
@@ -5,44 +5,44 @@
-    while (last > path && *last == '/')
+	last = path + strlen(path) - 1;
+	while (last > path && *last == '/') {
 	*last-- = '\0';
```
Missed indent here.

sergey-v commented on basenames.c:
```diff
@@ -5,44 +5,44 @@
-    } else {
+	}
+	else {
 	/* Remove any trailing slashes and final element. */
```
Why?

sergey-v commented on basenames.c:
```diff
@@ -5,44 +5,44 @@
 	return strdup(".");
-    } else {
+	}
+	else {
```
Why?

cromerc replied:
Why? Because I am human. I just fixed the indention in over 3,000 lines of code. If I missed some sorry. But thank you for pointing out one that I missed.

kdudka commented:
Could you please resubmit a minimal patch that just fixes the warning and nothing else?

We prefer not to apply unnecessary changes in order to ease history tracking and backporting patches for older releases of logrotate.

Thanks in advance!

cromerc commented:
Ok, I will submit the patch later tonight after I get out of the university.

kdudka commented:
Sounds good.  I am going to release ```3.10.0``` soon.
